### ISSUE TITLE
How do I use generated database types?

### ISSUE THREAD

**Issue by gergely-xyz** [nuxt-modules/supabase#150]
I would like to use the types generated with the supabase cli. #90  was closed as completed so I'm assuming this is possible, but I can't seem to make it work.

I've generated the types of my database:
`supabase gen types typescript --linked > supabase/schema.ts`

And added the following to my `nuxt.config.ts`:
```ts
import { Database } from './supabase/schema'
export default defineNuxtConfig({
// ...
  supabase: {
    client: {
      db: {
        schema: Database,
      },
    },
  },
// ...
}
```
The [documentation](https://supabase.nuxtjs.org/get-started#client) references [this](https://github.com/supabase/supabase-js/blob/master/src/lib/types.ts#L10) for available options.
I expected type hints to show up as shown [here](https://www.youtube.com/watch?v=7CqlTU9aOR4).

**Comment by ozum** [comment]
It would be a better DX if we can add this to the module configuration to be automatically used for every `useSupabaseClient` call.

```js
export default defineNuxtConfig({
  // ...
  supabase: {
    types: "~/types/database"
  }
}
```

Just for this purpose, I also created a PR (#160) to make auto imports optional, so I can create my own composable which adds types and reexports the client.

**/composables/useSupaClient.ts**
```ts
import type { Database } from "~/types/database";
export default () => useSupabaseClient<Database>();
```

Currently, I have to use a different name, because auto imported composables overwrite the local composables with same name.

Problem is, other developers may not know that there is a `useSupaClient` composable and use by mistake auto imported `useSupabaseClient` composable.

**Comment by zernonia** [comment]
I agree that adding types in config will improve DX by a huge factor. I wouldn't want to keep importing the same types over and over again in components/composables.
